fix(auto-mark-as-cleared): Update selector

michaelcmelton opened the pull request:
GitHub Issue (if applicable): #2799

Trello Link (if applicable): N/A

**Explanation of Bugfix/Feature/Modification:**
YNAB recently changed the classes and that broke this feature. Fix is to
update the selector utilized in the invoke function to keep up to date
with the current CSS classes of the cleared button.

Closes #2799

joshmadewell reviewed (changes requested):
i think this might have been fixed by the change that put our `MutationObserver` on `document.body` instead of `.layout` -- might be worth double checking this bug on main.

michaelcmelton commented:
I migrated this feature to addEmberToolkitHook as well, and removed the observe function. Verified this works for both single category and split transactions.

joshmadewell commented:
hmm, I'm not able to reproduce the original bug? my new transactions are getting marked as clear.

it looks like your code updates to make it so when you edit a transaction it's automatically marked as cleared but that's not what this feature should do. it should just pre-click the cleared symbol for you when you're adding a transaction.

i'm fine with updating this to use `addToolkitEmberHook` but we should be adding to `register/grid-add`

joshmadewell commented:
Ah I know why this is happening intermittently.

I guess I reverted the change which is now available in: https://github.com/toolkit-for-ynab/toolkit-for-ynab/pull/2819 which should fix this. Basically, our Mutation Observer just gets removed when you browse between budget->accounts and never gets reapplied. My guess is that `.ember-view.layout` is getting remounted now which is breaking our observer. `document.body` is always there so we're just going to attach to that.

michaelcmelton commented:
So this is obselete? I can delete my branch, yeah?

joshmadewell commented:
yeah, this was just a bigger issue and probably the reason for a few bugs in our board too :(
